### ISSUE TITLE
PackageManager#getLaunchIntentForPackage() returns null

### ISSUE THREAD

**Issue by screendriver** [robolectric/robolectric#747]
Since the release of robolectric 2.2 the metod getLaunchIntentForPackage() returns null. In older releases it worked fine.

``` java
Context context = Robolectric.application;
Intent activityIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
// activityIntent is null
```

**Comment by Miksu82** [comment]
Any updates? Seems to be still broken in version 3.1.2.

**Comment by joebowbeer** [comment]
Still broken in 3.1.4

**Comment by chris-hatton** [comment]
Any workaround for this?  I'm also stuck with an NPE in code which uses `getLaunchIntentForPackage`, but which I do not manage.

**Comment by Miksu82** [comment]
I do something like this

```
private void injectLaunchIntent() {
        Intent launchIntent = new Intent(Intent.ACTION_MAIN);
        launchIntent.addCategory(Intent.CATEGORY_LAUNCHER);
        launchIntent.setPackage("com.mydomain.mypackage");

        ResolveInfo resolveInfo = new ResolveInfo();
        ActivityInfo activityInfo = new ActivityInfo();
        activityInfo.packageName = "com.mydomain.mypackage";
        activityInfo.name = "MyMainActivity";
        resolveInfo.activityInfo = activityInfo;

        Context context = ShadowApplication.getInstance().getApplicationContext();
        ShadowPackageManager packageManeger = shadowOf(context.getPackageManager());
        packageManeger.addResolveInfoForIntent(launchIntent, resolveInfo);
}
```

**Comment by jongerrish** [comment]
Looks like this has been implemented, can you try with Robolectric 4.0

If there is still an issue and you think this was closed by mistake feel free to reopen.
